### ISSUE TITLE
Models Take Forever to Load.

### ISSUE THREAD

**Issue by SaarM2005** [lllyasviel/Fooocus#1471]
I have an Intel(R) Core(TM) i5-9500F CPU, 16.0GB Ram, and an NVIDIA GeForce GTX 1660.
It usually takes me around 250 seconds to generate an image on extreme speed, most of the time is spent on loading the model.
I have tried adding more swappable files space but it changed nothing.
After looking at the log, it seems it fails to load config keys. Is this related?

**Full Console Log**

D:\Fooocus>.\python_embeded\python.exe -s Fooocus\entry_with_update.py
Already up-to-date
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py']
Failed to load config key: {"path_checkpoints": "D:\\AI\\Fooocus\\models\\checkpoints"} is invalid or does not exist; will use {"path_checkpoints": "../models/checkpoints/"} instead.
Failed to load config key: {"path_loras": "D:\\AI\\Fooocus\\models\\loras"} is invalid or does not exist; will use {"path_loras": "../models/loras/"} instead.
Failed to load config key: {"path_embeddings": "D:\\AI\\Fooocus\\models\\embeddings"} is invalid or does not exist; will use {"path_embeddings": "../models/embeddings/"} instead.
Failed to load config key: {"path_vae_approx": "D:\\AI\\Fooocus\\models\\vae_approx"} is invalid or does not exist; will use {"path_vae_approx": "../models/vae_approx/"} instead.
Failed to load config key: {"path_upscale_models": "D:\\AI\\Fooocus\\models\\upscale_models"} is invalid or does not exist; will use {"path_upscale_models": "../models/upscale_models/"} instead.
Failed to load config key: {"path_inpaint": "D:\\AI\\Fooocus\\models\\inpaint"} is invalid or does not exist; will use {"path_inpaint": "../models/inpaint/"} instead.
Failed to load config key: {"path_controlnet": "D:\\AI\\Fooocus\\models\\controlnet"} is invalid or does not exist; will use {"path_controlnet": "../models/controlnet/"} instead.
Failed to load config key: {"path_clip_vision": "D:\\AI\\Fooocus\\models\\clip_vision"} is invalid or does not exist; will use {"path_clip_vision": "../models/clip_vision/"} instead.
Failed to load config key: {"path_fooocus_expansion": "D:\\AI\\Fooocus\\models\\prompt_expansion\\fooocus_expansion"} is invalid or does not exist; will use {"path_fooocus_expansion": "../models/prompt_expansion/fooocus_expansion"} instead.
Failed to load config key: {"path_outputs": "D:\\AI\\Fooocus\\outputs"} is invalid or does not exist; will use {"path_outputs": "../outputs/"} instead.
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.1.850
Running on local URL:  http://127.0.0.1:7865

To create a public link, set `share=True` in `launch()`.
Total VRAM 6144 MB, total RAM 16327 MB
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce GTX 1660 : native
VAE dtype: torch.float32
Using pytorch cross attention
Refiner unloaded.
model_type EPS
UNet ADM Dimension 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_l.logit_scale'}
Base model loaded: D:\Fooocus\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors
Request to load LoRAs [['sd_xl_offset_example-lora_1.0.safetensors', 0.1], ['None', 1.0], ['None', 1.0], ['None', 1.0], ['None', 1.0]] for model [D:\Fooocus\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors].
Loaded LoRA [D:\Fooocus\Fooocus\models\loras\sd_xl_offset_example-lora_1.0.safetensors] for UNet [D:\Fooocus\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors] with 788 keys at weight 0.1.
Fooocus V2 Expansion: Vocab with 642 words.
Fooocus Expansion engine loaded for cpu, use_fp16 = False.
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
App started successful. Use the app with http://127.0.0.1:7865/ or 127.0.0.1:7865
Enter LCM mode.
[Fooocus] Downloading LCM components ...
[Parameters] Adaptive CFG = 1.0
[Parameters] Sharpness = 0.0
[Parameters] ADM Scale = 1.0 : 1.0 : 0.0
[Parameters] CFG = 1.0
[Parameters] Seed = 5619124201367032379
[Parameters] Sampler = lcm - lcm
[Parameters] Steps = 8 - 8
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
Request to load LoRAs [['sd_xl_offset_example-lora_1.0.safetensors', 0.1], ['None', 1.0], ['None', 1.0], ['None', 1.0], ['None', 1.0], ('sdxl_lcm_lora.safetensors', 1.0)] for model [D:\Fooocus\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors].
Loaded LoRA [D:\Fooocus\Fooocus\models\loras\sd_xl_offset_example-lora_1.0.safetensors] for UNet [D:\Fooocus\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors] with 788 keys at weight 0.1.
Loaded LoRA [D:\Fooocus\Fooocus\models\loras\sdxl_lcm_lora.safetensors] for UNet [D:\Fooocus\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors] with 788 keys at weight 1.0.
Requested to load SDXLClipModel
Loading 1 new model
unload clone 1
[Fooocus] Processing prompts ...
[Fooocus] Preparing Fooocus text #1 ...
[Prompt Expansion] A cat, full color, cinematic, sharp focus, romantic, elegant, highly detailed, best, vivid, modern, rational, epic, coherent, magical, artistic, divine, scenic, thought, magic, mystical, amazing, symmetry, intricate, elaborate, colorful, light, stunning, creative, awesome, illuminated, wonderful, radiant, very inspirational, great, attractive
[Fooocus] Encoding positive #1 ...
[Parameters] Denoising Strength = 1.0
[Parameters] Initial Latent shape: Image Space (896, 1152)
Preparation time: 7.29 seconds
Using lcm scheduler.
[Sampler] refiner_swap_method = joint
[Sampler] sigma_min = 0.39970141649246216, sigma_max = 14.614640235900879
Requested to load SDXL
Loading 1 new model
loading in lowvram mode 3166.1985425949097
[Fooocus Model Management] Moving model(s) has taken 177.27 seconds
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8/8 [00:55<00:00,  6.93s/it]
Requested to load AutoencoderKL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 3.87 seconds
Image generated with private log at: D:\Fooocus\Fooocus\outputs\2023-12-17\log.html
Generating and saving time: 246.75 seconds
Total time: 255.21 seconds

**Comment by mashb1t** [comment]
please refer to https://github.com/lllyasviel/Fooocus/issues/1401 as Fooocus automatically switches from normal vram mode to low vram mode on your machine

**Comment by SaarM2005** [comment]
I followed the steps and changed my swap files to 40gb on my ssd drive, but it seems it just takes longer (385.58 overall, 208.24 to move model).
Can changing it manually to normal vram mode help? if so how do I do it?

**Comment by mashb1t** [comment]
@SaarM2005 sure, you can always enforce a specific mode. For normal vram the argument is `--always-normal-vram`, see https://github.com/lllyasviel/Fooocus/blob/main/ldm_patched/modules/args_parser.py#L98
Hope that helps!

**Comment by SaarM2005** [comment]
I dont really know how to do it because I'm very new to python, but do you think it can help? the big delay is specifically between the lines "Loading 1 new model" and "loading in lowvram mode 3166.1985425949097".

**Comment by mashb1t** [comment]
@SaarM2005 your GPU is the limiting factor. If you see low usage of VRAM in lowvram mode then it might be benificial to use normalvram mode instead, but this might not work.
You can simply add `--always-normal-vram` after the  entry_with_update.py call in your run.bat file, same line, before pause.

`.\python_embeded\python.exe -s Fooocus\entry_with_update.py --always-normal-vram`

**Comment by SaarM2005** [comment]
It's still saying "loading in lowvram mode 3166.1985425949097"

**Comment by SaarM2005** [comment]
Well, thanks anyway!

**Comment by Pareash** [comment]
I have similar issue it take forever to load and end up having errored message. 
I have a AMD Ryzen 5 4600H with Radeon Graphics (512 MB) 3.00 GHz  CPU, 8 GB Ram, and an NVIDIA GeForce GTX 1650.
Os: Windows 11

Please help i am trying since 2 weeks. Not single image generated. It always stuck at this stage. 
I tried Systam swap but still not working. 

Anything wrong with the hardware?

**Comment by mashb1t** [comment]
@Pareash i'm sorry, your GPU isn't meeting the minimal requirements necessary to run Fooocus, see https://github.com/lllyasviel/Fooocus?tab=readme-ov-file#minimal-requirement.
You can try `--always-cpu` though, but i hope you have an SSD with 30-40GB free space, as your RAM (8GB) will be filled in no time 😐

**Comment by Pareash** [comment]
@mashb1t Thanks for the reply
Yes i have free space of 200 gb in SSD 
please share steps for --always-cpu

**Comment by mashb1t** [comment]
@Pareash there basically is only one step, to modify your entrypoint (run.bat or call in terminal directly, depending on which OS you're using): 
```
.\python_embeded\python.exe -s Fooocus\entry_with_update.py --always-cpu
```

**Comment by WhiteShed** [comment]
Hello. I have same problem. Models downloading forever. Also when i'm trying to download from huggingface, it takes forever.
then fail(
when i run bat file from administrator. there is mistake 
":\Windows\System32>.\python_embeded\python.exe -s Fooocus\entry_with_update.py
Системе не удается найти указанный путь.

C:\Windows\System32>pause
Для продолжения нажмите любую клавишу"

so i run it double click. it worked after i manually download it through console T_T
then after first start there is this text:
"Already up-to-date
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.5.5
[Cleanup] Attempting to delete content of temp dir C:\Users\(****)\AppData\Local\Temp\fooocus
[Cleanup] Cleanup successful
Downloading: "https://huggingface.co/lllyasviel/fav_models/resolve/main/fav/juggernautXL_v8Rundiffusion.safetensors" to C:\Users\KitsuneRi\Downloads\Fooocus_win64_2-5-0\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors

  0%|                                                                                      | 0.00/6.62G [00:00<?, ?B/s]"
How can i solve this problem T_T
